spec/models/portfolio_spec.rb

```diff
@@ -130,4 +130,53 @@
       expect(Portfolio.taggable?).to be_truthy
     end
   end
+
+  context 'callbacks' do
+    before { subject.run_callbacks :create }
+
+    it 'adds limited keys to metadata' do
+      expect(subject.metadata.keys).to_not include(AncillaryMetadata::NON_METADATA_ATTRIBUTES)
+    end
+
+    it 'adds staticsitcs' do
+      expect(subject.metadata['statistics']).to include(
+        'approval_processes' => 0,
+        'portfolio_items'    => 0,
+        'shared_groups'      => 0
+      )
+    end
+  end
+
+  describe '#update_metadata' do
+    context 'ancillary_metadata instance does not exist' do
+      it 'creates ancillary_metadata instance but does not save it' do
+        expect(subject).to receive(:build_ancillary_metadata).and_call_original
+        expect(subject).to receive(:update_ancillary_metadata)
```

Comment:
We're kinda doing the same thing here, stubbing the object under test, but it's a lot harder to avoid because we're mixing in the functionality. I think I'm ok with it since we do have specs above that are testing the statistics functionality and the controller spec that tests it as well, but this whole `#update_metadata` group of specs is probably something that we could look at to be refactored at a later date.